ampligraph/datasets/oneton_adapter.py

```diff
@@ -0,0 +1,441 @@
+
+        if use_filter and self.filter_mapping is None:
+            msg = 'Cannot set `use_filter=True` if a filter has not been set in the adapter. '
+            logger.error(msg)
+            raise ValueError(msg)
+
+        if not self.low_memory:
+
+            if not self.verify_outputs(dataset_type, use_filter=use_filter, unique_pairs=unique_pairs):
+                # Verifies that onehot outputs are as expected given filter and unique_pair settings
+                msg = 'Generating one-hot outputs for {} [filtered: {}, unique_pairs: {}]'\
+                    .format(dataset_type, use_filter, unique_pairs)
+                logger.info(msg)
+                self.generate_outputs(dataset_type, use_filter=use_filter, unique_pairs=unique_pairs)
+
+            # Yield batches
+            for i in range(batches_count):
+
+                out = np.int32(X[(i * batch_size):((i + 1) * batch_size), :])
+                out_onehot = self.output_onehot[dataset_type][(i * batch_size):((i + 1) * batch_size), :]
+
+                yield out, out_onehot
+
+        else:
+            # Low-memory, generate one-hot outputs per batch on the fly
+            if use_filter:
+                output_dict = self.filter_mapping
+            else:
+                output_dict = self.output_mapping
+
+            # Yield batches
+            for i in range(batches_count):
+
+                out = np.int32(X[(i * batch_size):((i + 1) * batch_size), :])
+                out_onehot = np.zeros(shape=[out.shape[0], len(self.ent_to_idx)], dtype=np.int32)
+
+                for j, x in enumerate(out):
+                    indices = output_dict.get((x[0], x[1]), [])
+                    out_onehot[j, indices] = 1
+
+                yield out, out_onehot
+
+    def get_next_batch_subject_corruptions(self, batch_size=-1, dataset_type='train', use_filter=True):
+        """Batch generator for subject corruptions.
+
+        To avoid multiple redundant forward-passes through the network, subject corruptions are performed once for
+        each relation, and results accumulated for valid test triples.
+
+        If there are no test triples for a relation, then that relation is ignored.
+
+        Use batch_size to control memory usage (as a batch_size*N tensor will be allocated, where N is number
+        of unique entities.)
+
+        Parameters
+        ----------
+        batch_size: int
+            Maximum batch size returned.
+        dataset_type: string
+            indicates which dataset to use
+        use_filter : bool
+            Flag to indicate whether to return the one-hot outputs are generated from filtered or unfiltered datasets
+
+        Returns
+        -------
+
+        test_triples : nd-array, shape=[batch_size, 3]
+            A set of triples from the dataset type specified, that include the predicate currently returned in batch.
+        batch_triples : nd-array of shape (N, 3), where N is number of unique entities.
```

Comment:
fix the doc string (return sizes)

Reply:
Docstring details were actually incorrect, so : fixed